### PR TITLE
Updates to calendar and arrow icons

### LINKED ISSUE #2244: Add Redo and Undo to Nimble Icon Tokens
Please add and update the following icons to the Token Library

@jattasNI @fredvisser 

**Add:**

**arrow-u-undo_16x16** Tag: Undo, Revert, Back
Image

**arrow-u-redo_16x16** Tag: Redo, Revert, Forward
Image

**link-right-arrow_16x16**  Tag: Link, Open
Image

**Update:**

**anchor-box-up-right-arrow_16x16**
Image

### LINKED ISSUE #2256: Add Icons to Token Library for Soliton (calendar variants)
# 🎨 Icon Request

@jattasNI @fredvisser  Please add the following to the token library for the Soliton team:

**Remove** calendar-date-time_16x16 (if it currently exists in the library) and replace with **calendar-one-square-day_16x16**
Image

**Add**

**calendar-one-hollow-square-today_16x16**
Image

**calendar-top-rectangle-week_16x16**
Image

**calendar-filled-rectangle-year_16x16**
Image

**calendar-clock-scheduling-assistant_16x16**
Image

**calendar-three-rectangles-agenda_16x16**
Image

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 🤨 Rationale

Fixes #2244
Fixes #2256

## 👩‍💻 Implementation

Added:
1. `nimble-icon-arrow-right-thin`
1. `nimble-icon-arrow-u-left`
1. `nimble-icon-arrow-u-right`
1. `nimble-icon-calendar-clock`
1. `nimble-icon-calendar-day-outline`
1. `nimble-icon-calendar-week`
1. `nimble-icon-calendar-lines`
1. `nimble-icon-calendar-rectangle`

Replaced svg for:
- `nimble-icon-calendar-day`
- `nimble-icon-arrow-up-right-from-square`

Renamed:
- `nimble-icon-calendar` --> `nimble-icon-calendar-days`
- `nimble-icon-arrow-u-rotate-left` --> `nimble-icon-arrow-u-up`

Updated the Storybook icon table so that the first column is narrower, leaving more room in the other columns to avoid truncating text.

## 🧪 Testing

N/A

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json
Comment (m-akinc) [inline change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:1]: This is a breaking change, and there are [several usages](https://codesearch.natinst.com/search/text?q=(nimble-icon-calendar-day%7Cnimble-icon-calendar(%5B%5E-%5D%7C%24))&fold_case=auto&regex=true&context=true) that will need to be fixed in order to uptake this. Specifically, LabManagement, RoutineManager, SystemLinkShared, and even [some dashboard app in the Tools repo](https://dev.azure.com/ni/DevCentral/_git/Tools?path=/report/dashboard/frontend_v2/src/app/systems/run-claims/run-claims.component.html&version=GC79cbf6e0c846859ef2325726c472872b3ec0b57e&line=18&lineEnd=19&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) will be broken.

**Comment by m-akinc** [comment]
@msmithNI do you mind buddying?

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (rajsite) [inline packages/nimble-components/src/icon-base/tests/icon-metadata.ts:115]: Can you make sure the names are following the naming guidelines?
https://github.com/ni/nimble/blob/main/packages/nimble-tokens/CONTRIBUTING.md#icon-naming

High-level concerns:

- These names don't seem like what I expect from font awesome, but I did not search so may help to reference if used
- Names like `IconCalendarThreeRectanglesAgenda` seem to be mixing both "descriptive names" (calendar three rectangles) and "Metaphors" (agenda)

May be useful to update contributing docs for icon names to encourage linking to the relevant font awesome names in PR description or PR comments.

Comment (m-akinc) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:115]: @NIbokeefe wanted these specific names. It seems like whoever decides on the naming scheme is intentionally adding metaphors to the names. He already expressed some concern about our element names not matching the SVG filenames exactly, so I think we need to have a discussion.

Comment (rajsite) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:115]: Throwing up the @fredvisser bat signal 🦇🔦

File: change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json
Comment (rajsite) [inline change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:1]: I'll make a separate comment, but to accelerate this workflow maybe we should create a new issue template specifically for icons that covers some points:
- Links to the naming scheme documentation
- Adds fields so that the referenced font awesome name can be linked and metaphors pulled from or the alternate name if a font awesome one is not available
- Includes additional metaphors that are needed
- Does not move out of triage / move to ready until those are established and can be merged mechanically

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (fredvisser) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:115]: I understand Brandon's concerns, but we need to use the font awesome naming scheme, or this will descend into chaos. I.e.

IconCalendarClockSchedulingAssistant → IconCalendarClock
IconCalendarOneSquareDay → IconCalendarDay
IconCalendarOneHollowSquareToday → IconCalendarToday
IconCalendarThreeRectanglesAgenda → IconCalendarLines
IconCalendarTopRectangleWeek →  IconCalendarWeek
IconCalendar/IconCalendarMonth → IconCalendarDays (we could defer this to avoid a breaking change)

IconArrowURedo → IconArrowRotateRight
IconArrowUUndo → IconArrowRotateLeft

IconLinkRightArrow → IconArrowRight

Comment (m-akinc) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:115]: There's some stuff we need to get consensus on.

1. Fred, the names you propose are a mix of descriptions and metaphors, for example `IconCalendarDay` (metaphor) vs `IconCalendarOneSquare` (description). I can see the argument for the metaphor-based name, since there really shouldn't be any other use case/metaphor for that icon, but it's still inconsistent with our naming pattern so far (I think).
2. If `IconLinkRightArrow` is the icon we'll use specifically for indicating links, should we reserve it for that specific purpose and use some other variation of a right-pointing arrow for other contexts? The same question goes for `IconAnchorBoxUpRightArrow`. My concern is that an icon that has established itself to mean one thing in 99% of use cases may be confusing when used in another context 1% of the time.
3. We should pick either "link" or "anchor" and consistently use that term (in icon names, at least).
4. Fred, your recommended names `IconArrowRotateRight` and `IconArrowRotateLeft` are too ambiguous and the former conflicts with an existing icon name. We also have an existing icon `IconArrowURotateLeft` which is not a great name, because (1) the arrow head points up, not to the left, and (2) the arrow makes a clockwise turn, which is not what "rotate left" would imply. I propose we rename the existing icon `IconArrowUUp`, and then our new ones logically follow as `IconArrowULeft` and `IconArrowURight`. This naming scheme assumes we'll never need both "isomers" of the icon, e.g.
image

Comment (fredvisser) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:115]: This is the beauty of our [icon naming guidance](https://github.com/ni/nimble/blob/main/packages/nimble-tokens/CONTRIBUTING.md#icon-naming):

>Use [Font Awesome](https://fontawesome.com/) names for icons. If Font Awesome doesn't contain an equivalent icon, minimize the use of metaphors; choose descriptive and unambiguous names instead.

Font Awesome [has](https://fontawesome.com/search?q=calendar&o=r):
- calendar-clock
- calendar-day
- calendar-lines
- calendar-week
- calendar-days

…so those are all good. These aren't metaphors so much as clear descriptions of the icons. I.e. It's not _agenda_ or _schedule_.

And for the link icons, font awesome also [has those](https://fontawesome.com/search?q=link&o=r):
- arrow-up-right-from-square
- arrow-right

…and we shouldn't include link or anchor in the icon name for an arrow. We can expand our icon documentation elsewhere, but this is not the place to bake it in.

Screenshot 2024-07-11 at 2 01 10 PM 🤦‍♂️

And yeah, for #4 - I like your proposal - 👍

File: change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json
Comment (m-akinc) [reply change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:1]: Filed tech debt #2263

Comment (rajsite) [inline change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:1]: @m-akinc fyi some of the armstrong folks were also interested in the status of the PR. Still in draft / has open comments / reviewers not reset so assuming still in work and not waiting on reviews.

Comment (m-akinc) [reply change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:1]: Just now got the names finalized and let Malcolm know it's ready for buddy.

**Review by msmithNI — APPROVED** [review]

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (rajsite) [inline packages/nimble-components/src/icon-base/tests/icon-metadata.ts:54]: Do we still need this icon since the link pattern has changed? I think the tags listed are incorrect now right @fredvisser ? What are the other use cases for keeping it?

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (rajsite) [inline packages/nimble-components/src/icon-base/tests/icon-metadata.ts:130]: I may have missed it but I don't see [calendar today](https://fontawesome.com/search?q=calendar%20today&o=r) in font awesome so this seems like a different `calendar-day` style.

What about `calendar-day-outline` with a tag of `today` as the metaphor? @fredvisser

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (rajsite) [inline packages/nimble-components/src/icon-base/tests/icon-metadata.ts:129]: Is there and intended metaphor for this icon's usage? From the initial issue with name `calendar-filled-rectangle-year_16x16` I'd guess "year" but I also don't know how well this icon represents a "year". If anything this looks like a "month" picker based on the other visuals for day and week that cover parts of the calendar.

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (rajsite) [inline packages/nimble-components/src/icon-base/tests/icon-metadata.ts:123]: Are there any tags to differentiate usage of `day` vs `days`?

**Review by rajsite — APPROVED** [review]

File: change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json
Comment (rajsite) [inline change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:3]: I don't think folks looking at the git history / PR titles care as much about the number of icons vs what the icons are at a high level

```suggestion
  "comment": "Updates to calendar and arrow icons. Clients using \"nimble-icon-calendar\" or \"nimble-icon-arrow-u-rotate-left\" must instead use \"nimble-icon-calendar-days\" and \"nimble-icon-arrow-u-up\" (respectively).",
```

File: change/@ni-nimble-tokens-d11f8624-58cd-4e30-92a8-670f7f40b023.json
Comment (rajsite) [inline change/@ni-nimble-tokens-d11f8624-58cd-4e30-92a8-670f7f40b023.json:3]:
```suggestion
  "comment": "Updates to calendar and arrow icons. Clients using \"nimble-icon-calendar\" or \"nimble-icon-arrow-u-rotate-left\" must instead use \"nimble-icon-calendar-days\" and \"nimble-icon-arrow-u-up\" (respectively).",
```

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (m-akinc) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:129]: Yes, I guess it should list the metaphor "year", but I agree that it seems ambiguous. @NIbokeefe can you confirm? Year? Month? Both?

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (m-akinc) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:123]: Paging Dr. @NIbokeefe

File: change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json
Comment (jattasNI) [inline change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:6]: Clients will mostly be using the dependent libraries nimble-angular or NimbleBlazor and it'd be good to signal the breaking change there too.
```suggestion
  "dependentChangeType": "major"
```

File: change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json
Comment (jattasNI) [reply change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:1]: Nice job with the change message! Reminder to either update the SLE apps yourself or post in the UI WG channel for visibility.

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (jattasNI) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:129]: Only barely related but I came across this calendar icon set that people were raving about on the internet. They also had to stretch to extend the metaphor to "year".

https://www.reddit.com/r/DesignPorn/comments/1azciu3/microsoft_to_do_repeat_icons/

image

**Review by jattasNI — APPROVED** [review]
Interested in the answers to Milan's questions but otherwise looking good

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (NIbokeefe) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:129]: > Yes, I guess it should list the metaphor "year", but I agree that it seems ambiguous. @NIbokeefe can you confirm? Year? Month? Both?

Comment (NIbokeefe) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:129]: > Yes, I guess it should list the metaphor "year", but I agree that it seems ambiguous. @NIbokeefe can you confirm? Year? Month? Both?

Comment (NIbokeefe) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:129]: Screenshot 2024-07-16 at 10 23 13 AM

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (NIbokeefe) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:123]: Not currently.  Right now I've only gotten a request for calendar "day." And the closest related one is "today"

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (NIbokeefe) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:129]: I agree, I didn't like this one for year either but they are so small it wasn't possible at add 365 or anything like that.  It was Madhamitas design and I attempted a couple other versions. Went over them with Fred and let him be the 'D' which one.

Screenshot 2024-07-16 at 10 11 13 AM

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (fredvisser) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:54]: Yep - we still need it, because we added it to the [standalone link](https://stratus-storybook.ni.dev/iframe.html?viewMode=docs&id=patterns-links--docs&args=#standalone-links) pattern

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (fredvisser) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:130]: Yeah, I waffled between those 2 naming schemes - I'm good with `calendar-day-outline` and tag of `today` as the metaphor

File: packages/nimble-components/src/icon-base/tests/icon-metadata.ts
Comment (m-akinc) [reply packages/nimble-components/src/icon-base/tests/icon-metadata.ts:129]: I'm updating the metaphors for `IconCalendarDay` (removing "time" and "schedule", so now just "date"), `IconCalendarDays` (adding "month" alongside existing "schedule"), and `IconCalendarRectangle` (added "year"). I am not adding tags for words that are already in the name of the icon (e.g. "day" or "week").

File: package-lock.json
Comment (m-akinc) [inline package-lock.json:1]: Nothing I changed should have required a lock file change, but my build failed, and running `npm i` locally created this change, so..

File: change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json
Comment (rajsite) [inline change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:1]: @m-akinc looks ready, can you double check the PR description is accurate?

Comment (m-akinc) [reply change/@ni-nimble-components-56124785-d76b-4bd0-b330-b457eb4ff9a0.json:1]: Yep, now up-to-date.

**Pull request merged**
